src/support.rs

```diff
@@ -0,0 +1,87 @@
+use crate::palloc::{Inner, InternalAsValue, ToInternal};
+use pgx::*;
+
+// FIXME: This rough translation from C to Rust is _untested_ and probably broken
+
+#[pg_extern(immutable, strict)]
+#[search_path(@extschema@)]
+pub unsafe fn make_call_subquery_support(input: Internal) -> Internal {
+    make_call_subquery_support_inner(input.to_inner()).internal()
+}
+
+pub unsafe fn make_call_subquery_support_inner(
+    input: Option<Inner<*mut pg_sys::Node>>,
+) -> Inner<*mut pg_sys::Node> {
+    let input: *mut pg_sys::Node = input.unwrap().cast();
+    if !pgx::is_a(input, pg_sys::NodeTag_T_SupportRequestSimplify) {
+        return (0 as *mut pg_sys::Node).into();
+    }
+
+    let req: *mut pg_sys::SupportRequestSimplify = input.cast();
+
+    let root = (*req).root;
+
+    if root.is_null() {
+        return (0 as *mut pg_sys::Node).into();
+    }
+
+    /*
+     * This prevents recursion of this optimization when the subselect is
+     * planned
+     */
+    if (*root).query_level > 1 {
+        return (0 as *mut pg_sys::Node).into();
+    }
+
+    let expr = (*req).fcall;
+
+    let original_args = PgList::<pg_sys::Node>::from_pg((*expr).args);
+
+    /* Check that these are expressions that don't reference
+    any vars, i.e. they are constants or expressions of constants */
+    if !original_args
+        .iter_ptr()
+        .all(|arg| arg_can_be_put_into_subquery(arg))
+    {
+        return (0 as *mut pg_sys::Node).into();
```

Comment:
on a pure rustic-ness level this might be better written as
```suggestion
        return ptr::null_mut::<pg_sys::Node>().internal();
```

Reply:
Thanks for this suggestion! I have adopted it.